src/MICore/PlatformUtilities.cs

```diff
@@ -0,0 +1,149 @@
+            {
+                _runtimePlatform = CalculateRuntimePlatform();
+            }
+            return _runtimePlatform;
+        }
+
+#if !CORECLR
+        [DllImport("libc")]
+        static extern int uname(IntPtr buf);
+
+        private static RuntimePlatform GetUnixVariant()
+        {
+            IntPtr buf = Marshal.AllocHGlobal(8192);
+            try
+            {
+                if (uname(buf) == 0)
+                {
+                    string os = Marshal.PtrToStringAnsi(buf);
+                    if (String.Equals(os, "Darwin", StringComparison.Ordinal))
+                    {
+                        return RuntimePlatform.MacOSX;
+                    }
+                }
+            }
+            finally
+            {
+                if (buf != IntPtr.Zero)
+                {
+                    Marshal.FreeHGlobal(buf);
+                }
+            }
+
+            return RuntimePlatform.Unix;
+        }
+#endif
+
+        private static RuntimePlatform CalculateRuntimePlatform()
+        {
+#if CORECLR
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return RuntimePlatform.Windows;
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return RuntimePlatform.MacOSX;
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return RuntimePlatform.Unix;
+            }
+            return RuntimePlatform.Unknown;
+#else
+            const PlatformID MonoOldUnix = (PlatformID)128;
+
+            switch (Environment.OSVersion.Platform)
+            {
+                case PlatformID.Win32NT:
+                    return RuntimePlatform.Windows;
+                case PlatformID.Unix:
+                case MonoOldUnix:
+                    // Mono returns PlatformID.Unix on OSX for compatibility
+                    return PlatformUtilities.GetUnixVariant();
+                case PlatformID.MacOSX:
+                    return RuntimePlatform.MacOSX;
+                default:
+                    return RuntimePlatform.Unknown;
+            }
+#endif
+        }
+
+        /*
+         * Is this Windows?
+         */
+        public static bool IsWindows()
+        {
+            return GetRuntimePlatform() == RuntimePlatform.Windows;
+        }
+
+        /*
+         * Is this OS X?
+         */
+        public static bool IsOSX()
+        {
+            return GetRuntimePlatform() == RuntimePlatform.MacOSX;
+        }
+
+        /*
+         * Is this Linux?
+         */
+        public static bool IsLinux()
+        {
+            return GetRuntimePlatform() == RuntimePlatform.Unix;
+        }
+
+        // Abstract API call to add an environment variable to a new process
+        public static void AddToEnvironment(this ProcessStartInfo processStartInfo, string key, string value)
```

Comment:
I might suggest 'Set' rather than 'Add' here since Add often implies failing if it already exists.